onnxruntime/core/graph/graph_proto_serializer.h

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/graph/graph_viewer.h"
+
+namespace onnxruntime {
+
+class GraphProtoSerializer {
+ public:
+  GraphProtoSerializer(const GraphViewer* graph_view) : graph_viewer_(graph_view) {}
+
+  void ToProto(ONNX_NAMESPACE::GraphProto& graph_proto, bool include_initializer);
+
+ private:
+  const GraphViewer* graph_viewer_;
```

Comment:
What's the reason to store this as a member? Can ToProto be a static function that takes GraphViewer as a const-ref? This will make it easier for clients to use this functionality.

Reply:
we need to invoke this through shared library provider API, somehow i got error when pass with const ref. but i can double check.